### ISSUE TITLE
Fragpipe params parsing

### ISSUE THREAD

**Issue by rodvrees** [Proteobench/ProteoBench#407]
In the fragpipe.workflow file, from which fragpipe settings are parsed, default values are written even for tools that are unused in the current workflow. For example, when DIA quant is done with DIA-NN. The fragpipe.workflow file will still contain the following part, even though ionquant is not used: 

> ionquant.excludemods=
> ionquant.heavy=
> ionquant.imtol=0.05
> ionquant.ionfdr=0.01
> ionquant.light=
> ionquant.locprob=0.75
> ionquant.maxlfq=1
> ionquant.mbr=0
> ionquant.mbrimtol=0.05
> ionquant.mbrmincorr=0
> ionquant.mbrrttol=1
> ionquant.mbrtoprun=10
> ionquant.medium=
> ionquant.minfreq=0
> ionquant.minions=1
> ionquant.minisotopes=1
> ionquant.minscans=3
> ionquant.mztol=10
> ionquant.normalization=1
> ionquant.peptidefdr=1
> ionquant.proteinfdr=1
> ionquant.requantify=1
> ionquant.rttol=0.4
> ionquant.run-ionquant=true
> ionquant.tp=0
> ionquant.uniqueness=0
> ionquant.use-labeling=false
> ionquant.use-lfq=true
> ionquant.writeindex=0

Note however that ionquant is not used because of the following: 
> quantitation.run-label-free-quant=false

found further in the workflow file. Quantification is then done, but by DIA-NN (note that run-dia-nn == true):

> diann.fragpipe.cmd-opts=
diann.generate-msstats=true
diann.heavy=
diann.library=
diann.light=
diann.medium=
diann.q-value=0.01
diann.quantification-strategy=3
diann.quantification-strategy-2=QuantUMS (high accuracy)
diann.run-dia-nn=true
diann.run-dia-plex=false
diann.run-specific-protein-q-value=false
diann.unrelated-runs=false
diann.use-predicted-spectra=false

So in essence, ionquant params are unused (default ones are written but ignored because of run-label-free-quant == false, but are currently parsed and seen as used params anyway by ProteoBench, and DIA-NN params aren't.

In conclusion: the fragpipe params parser should reflect which parts of the workflow are actually used and which ones aren't, because otherwise parameters will be parsed that are actually not reflective of the actual workflow.

Workflow file from which I pasted excerpts:
[fragpipe.txt](https://github.com/user-attachments/files/17244784/fragpipe.txt)

P.S. A lot of relevant settings remain unparsed, too
